ibis/backends/impala/tests/snapshots/test_sql/test_nested_join_base/out.sql

```diff
@@ -7,7 +7,7 @@ WITH `t1` AS (
     1
 )
 SELECT
-  `t5`.`uuid`,
+  `t3`.`uuid`,
```

Comment:
This looks incorrect to me. Check out the expression:


```python
    t = ibis.table(dict(uuid="string", ts="timestamp"), name="t")
    counts = t.group_by("uuid").size()
    max_counts = counts.group_by("uuid").aggregate(max_count=lambda x: x[1].max())
    result = max_counts.left_join(counts, "uuid").select(counts)
```

The `uuid` column in the final `.select` call is coming from the **right** side of the join, so it should remain `t5.uuid` I think.

Reply:
The right side `Field` cannot dereference to the left side field because this is a left outer join, so the join key values for a given row may be different.

Reply:
The following is the repr (after setting `ibis.options.repr.show_variables = True`):

```
t := UnboundTable: t
  uuid string
  ts   timestamp

counts := Aggregate[t]
  groups:
    uuid: t.uuid
  metrics:
    CountStar(t): CountStar(t)

max_counts := Aggregate[counts]
  groups:
    uuid: counts.uuid
  metrics:
    max_count: Max(counts['CountStar(t)'])

JoinChain[max_counts]
  JoinLink[left, counts]
    max_counts.uuid == counts.uuid
  values:
    uuid:         counts.uuid
    CountStar(t): counts['CountStar(t)']
```

Isn't this the expected expr?

Reply:
You're right! I misread. Resolved.